onedal/neighbors/neighbors.py

```diff
@@ -568,14 +575,17 @@ def _onedal_predict(self, model, X, params, queue):
 
             return predict_alg(**params).compute(X, model)
 
-        policy = _get_policy(queue, X)
+        policy = self._get_infer_policy(queue, X)
+        X = _convert_to_supported(policy, X)
         backend = _backend.neighbors.regression if gpu_device \
             else _backend.neighbors.search
 
         if hasattr(self, '_onedal_model'):
             model = self._onedal_model
         else:
             model = self._create_model(backend)
+        if gpu_device and "responses" not in params["result_option"]:
+            params["result_option"] += "|responses"
```

Comment:
Move it to `_get_onedal_params` please.

Reply:
It is dependent on gpu_device which depends on queue

Reply:
> It is dependent on gpu_device which depends on queue

Function `_get_onedal_params` also have `queue` as an arg.

Reply:
It does not currently have queue as an arg. I can add it as an optional arg and move lines into function or I can leave it as is to keep consistency with _get_onedal_params across all algos.

Reply:
But in current master there is a similar param conditional outside of _get_onedal_params which is part of the reason I configure it the way it is: https://github.com/intel/scikit-learn-intelex/blob/master/onedal/neighbors/neighbors.py#L430-L431

Reply:
This is internal use func, there is no need saving consistency for these methods. It is not a public. So it could be and should extended in this case. Since it is on master you can keep it as is.
BTW kNN master is not good enough for the good code reference.

Reply:
And also I didn't get why we adding condition for GPU device?

Reply:
The addition was only necessary for distributed version so I think the latest commit most cleanly implements the logic and addresses your comments - https://github.com/ethanglaser/scikit-learn-intelex/blob/dev/eglaser-spmd-knn/onedal/spmd/neighbors/neighbors.py#L37-L41